### ISSUE TITLE
Add markdown lint validation for .md files to CI

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#8909]
[Markdown lint tool](https://github.com/markdownlint/markdownlint)
A tool to check markdown files and flag style issues.

We don't have a lot of md files, but if there is a tool to validate them, it makes sense to use it.

Current state:
```bash
$ sudo gem install mdl
$ mdl -g .
.github/CODE_OF_CONDUCT.md:64: MD034 Bare URL used
.github/CODE_OF_CONDUCT.md:120: MD034 Bare URL used
.github/CODE_OF_CONDUCT.md:128: MD034 Bare URL used
.github/CODE_OF_CONDUCT.md:129: MD034 Bare URL used
.github/CONTRIBUTING.md:4: MD001 Header levels should only increment by one level at a time
.github/CONTRIBUTING.md:13: MD012 Multiple consecutive blank lines
.github/CONTRIBUTING.md:8: MD013 Line length
.github/CONTRIBUTING.md:15: MD013 Line length
.github/CONTRIBUTING.md:1: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:4: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:6: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:14: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:16: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:19: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:20: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:23: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:24: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:25: MD022 Headers should be surrounded by blank lines
.github/CONTRIBUTING.md:14: MD025 Multiple top level headers in the same document
.github/ISSUE_TEMPLATE/bug_report.md:27: MD009 Trailing spaces
.github/ISSUE_TEMPLATE/bug_report.md:28: MD031 Fenced code blocks should be surrounded by blank lines
.github/ISSUE_TEMPLATE/bug_report.md:31: MD031 Fenced code blocks should be surrounded by blank lines
.github/ISSUE_TEMPLATE/bug_report.md:10: MD034 Bare URL used
.github/ISSUE_TEMPLATE/bug_report.md:41: MD034 Bare URL used
.github/ISSUE_TEMPLATE/bug_report.md:34: MD035 Horizontal rule style
.github/ISSUE_TEMPLATE/bug_report.md:38: MD035 Horizontal rule style
.github/ISSUE_TEMPLATE/feature_request.md:14: MD013 Line length
.github/ISSUE_TEMPLATE/feature_request.md:20: MD013 Line length
.github/ISSUE_TEMPLATE/feature_request.md:14: MD034 Bare URL used
.github/ISSUE_TEMPLATE/feature_request.md:20: MD034 Bare URL used
.github/PULL_REQUEST_TEMPLATE.md:17: MD034 Bare URL used
README.md:24: MD012 Multiple consecutive blank lines
README.md:70: MD012 Multiple consecutive blank lines
README.md:83: MD012 Multiple consecutive blank lines
README.md:33: MD013 Line length
README.md:161: MD013 Line length
README.md:52: MD022 Headers should be surrounded by blank lines
README.md:35: MD034 Bare URL used
README.md:57: MD034 Bare URL used
README.md:64: MD034 Bare URL used
README.md:68: MD034 Bare URL used
README.md:92: MD034 Bare URL used
README.md:96: MD034 Bare URL used
README.md:99: MD034 Bare URL used
README.md:104: MD034 Bare URL used
README.md:109: MD034 Bare URL used

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

**Comment by strkkk** [comment]
This can be even more useful for contribution repo, there are more docs in markdown
